pkg/client/openshift/buildconfig.go

```diff
@@ -101,6 +106,38 @@ func (b *buildConfig) TriggerBuild(bc *buildv1.BuildConfig, triggedBy string) (b
 	return true, nil
 }
 
+// TriggerFromFileBuild wil be called by kogito-cli when a build from file is performed.
+// When called a new build will be triggered with the request kogito resource or a tgz file.
+func (b *buildConfig) TriggerFromFileBuild(namespace string, r io.Reader, options *buildv1.BinaryBuildRequestOptions) (*buildv1.Build, error) {
+
+	result := &buildv1.Build{}
+	buildName := fmt.Sprintf("%s%s", options.Name, "-builder")
+
+	// before upload the file, make sure that the build exist
+	err := b.waitForBuildConfig(5, 2*time.Second, func() (err error) {
+		if bc, err := b.client.BuildCli.BuildConfigs(namespace).Get(buildName, metav1.GetOptions{}); err != nil && errors.IsNotFound(err) {
+			log.Debugf("", bc.Namespace)
+			return err
+		}
```

Comment:
What if err is produced and `errors.IsNotFound(err) ` is false? This case doesn't seem to be handled, code will behave like there is no error.

Reply:
few lines above the error is being handled, should I move it to somewhere else?

Reply:
below*

Reply:
I see now, thanks. I overlooked `(err error)` declaration, binding err to the output.